File: example/kotlin/src/testFixtures/kotlin/com/some/pkg/typesmodule/fixtures/Builders.kt
package com.some.pkg.typesmodule.fixtures

import com.some.pkg.typesmodule.api.*

data class DateRangeDef(
    var from: String = "someValue",
    var to: String = "someValue",
)
fun dateRange(init: DateRangeDef.() -> Unit = {}): DateRange {
    val def = DateRangeDef().apply(init)
    return dateRangeCreate(
        from = dateCreate(def.from),
        to = dateCreate(def.to),
    )
}

File: example/typescript/test/TypesModule/Fixtures/Builders.ts
namespace TypesModule.Builder {
    export interface DateRangeDef {
        from?: string,
        to?: string,
    }
    export function dateRange(def?: DateRangeDef): DateRange {
        const from = def?.from ?? "someValue"
        const to = def?.to ?? "someValue"

        return CustomTypesMapper.dateRangeCreate(
            CustomTypesMapper.dateCreate(from),
            CustomTypesMapper.dateCreate(to),
        )
    }
}